x-pack/plugins/apm/common/license_check.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ILicense, LicenseType } from '../../licensing/common/types';
+
+function isActiveLicense(licenseType: LicenseType, license?: ILicense) {
+  return license && license.isActive && license.hasAtLeast(licenseType);
+}
+
+export function isActivePlatinumLicense(license?: ILicense) {
+  return isActiveLicense('platinum', license);
+}
+
+export function isActiveGoldLicense(license?: ILicense) {
+  return isActiveLicense('gold', license);
+}
```

Comment:
nit: Personally I think the duplication instead of abstraction is clearer but I'm fine with either

```suggestion
export function isActivePlatinumLicense(license?: ILicense) {
  return license && license.isActive && license.hasAtLeast('platinum');
}

export function isActiveGoldLicense(license?: ILicense) {
    return license && license.isActive && license.hasAtLeast('gold');
}
```

Reply:
I just thought if we add more validations we'd only have to do:

- isActiveLicense('basic', license)
- isActiveLicense('enterprise', license)

Let me know if you think it should be changed.

Reply:
Fine to leave it as-is 👍